core/common/src/main/java/alluxio/util/io/BufferUtils.java

```diff
@@ -51,6 +55,53 @@ public static int byteToInt(byte b) {
    * this direct buffer should be discarded. This is unsafe operation and currently a work-around to
    * avoid huge memory occupation caused by memory map.
    *
+   * @param buffer bytebuffer
+   */
+  public static synchronized void cleanDirectBuffer(ByteBuffer buffer) {
+    Preconditions.checkNotNull(buffer, "buffer is null");
+    Preconditions.checkArgument(buffer.isDirect(), "buffer isn't a DirectByteBuffer");
+    int javaVersion = CommonUtils.getJavaVersion();
+    if (javaVersion < 9) {
+      cleanDirectBufferJava8(buffer);
+    } else {
+      cleanDirectBufferJava11(buffer);
+    }
+  }
+
+  /**
+   * <p>
+   * Note: This calls the cleaner method on jdk 9+.
+   * See <a
+   * href="https://stackoverflow.com/questions/2972986/how-to-unmap-a-file-from-memory-mapped-
+   * using-filechannel-in-java"
+   * >more discussion</a>.
+   *
+   * @param buffer the byte buffer to be unmapped, this must be a direct buffer
+   */
+  private static synchronized void cleanDirectBufferJava11(ByteBuffer buffer) {
+    try {
+      if (sByteBufferCleanerMethod == null || sUnsafeClass == null) {
+        try {
+          sUnsafeClass = Class.forName("sun.misc.Unsafe");
+        } catch (Exception e) {
+          // jdk.internal.misc.Unsafe doesn't yet have an invokeCleaner() method,
+          // but that method should be added if sun.misc.Unsafe is removed.
+          sUnsafeClass = Class.forName("jdk.internal.misc.Unsafe");
+        }
+        sByteBufferCleanerMethod = sUnsafeClass.getMethod("invokeCleaner", ByteBuffer.class);
+        sByteBufferCleanerMethod.setAccessible(true);
+      }
+      Field theUnsafeField = sUnsafeClass.getDeclaredField("theUnsafe");
+      theUnsafeField.setAccessible(true);
+      Object theUnsafe = theUnsafeField.get(null);
+      sByteBufferCleanerMethod.invoke(theUnsafe, buffer);
+    } catch (Exception e) {
+      LOG.warn("Failed to unmap direct ByteBuffer: {}, error message: {}",
+          buffer.getClass().getName(), e.getMessage());
+    }
+  }
+
+  /**
```

Comment:
do we have a test for this?

Reply:
we have a test for the method. java 8 version or java 11 version will be tested depending on the jvm